src/libraries/System.IO.FileSystem.Watcher/tests/Args.FileSystemEventArgs.cs

```diff
@@ -8,86 +8,93 @@ namespace System.IO.Tests
+        [InlineData("D:\\", "foo.txt", "D:\\foo.txt")]
+        [InlineData("E:\\bar", "foo.txt", "E:\\bar\\foo.txt")]
+        public static void FileSystemEventArgs_ctor_DirectoryIsAbsolutePath_Windows(string directory, string name, string expectedFullPath)
         {
-            FileSystemEventArgs args = new FileSystemEventArgs(changeType, directory, name);
+            FileSystemEventArgs args = new FileSystemEventArgs(WatcherChangeTypes.All, directory, name);
 
-            Assert.Equal(changeType, args.ChangeType);
-            Assert.Equal(AppendDirectorySeparator(Directory.GetCurrentDirectory()) + name, args.FullPath);
-            Assert.Equal(name, args.Name);
+            Assert.Equal(expectedFullPath, args.FullPath);
         }
 
         [Theory]
-        [InlineData(WatcherChangeTypes.Changed, "bar", "foo.txt")]
-        [InlineData(WatcherChangeTypes.Changed, "bar\\baz", "foo.txt")]
-        [InlineData(WatcherChangeTypes.All, "bar", "foo.txt")]
-        [InlineData(WatcherChangeTypes.All, "bar\\baz", "foo.txt")]
-        public static void FileSystemEventArgs_ctor_DirectoryIsRelativePath(WatcherChangeTypes changeType, string directory, string name)
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [InlineData("/", "foo.txt", "/foo.txt")]
+        [InlineData("/bar", "foo.txt", "/bar/foo.txt")]
+        public static void FileSystemEventArgs_ctor_DirectoryIsAbsolutePath_Unix(string directory, string name, string expectedFullPath)
         {
-            FileSystemEventArgs args = new FileSystemEventArgs(changeType, directory, name);
+            FileSystemEventArgs args = new FileSystemEventArgs(WatcherChangeTypes.All, directory, name);
 
-            directory = AppendDirectorySeparator(directory);
-
-            Assert.Equal(changeType, args.ChangeType);
-            Assert.Equal(AppendDirectorySeparator(Directory.GetCurrentDirectory()) + directory + name, args.FullPath);
-            Assert.Equal(name, args.Name);
+            Assert.Equal(expectedFullPath, args.FullPath);
         }
 
-        [Fact]
-        public static void FileSystemEventArgs_ctor_Invalid_EmptyDirectory()
+        [Theory]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [InlineData("bar", "foo.txt")]
+        [InlineData("bar\\baz", "foo.txt")]
+        public static void FileSystemEventArgs_ctor_DirectoryIsRelativePath_Windows(string directory, string name)
         {
-            Assert.Throws<ArgumentException>(() => new FileSystemEventArgs((WatcherChangeTypes)0, "", "foo.txt"));
+            FileSystemEventArgs args = new FileSystemEventArgs(WatcherChangeTypes.All, directory, name);
+
+            directory = PathInternal.EnsureTrailingSeparator(directory);
+
+            Assert.Equal(PathInternal.EnsureTrailingSeparator(Directory.GetCurrentDirectory()) + directory + name, args.FullPath);
```

Comment:
Path.Combine will ensure the trailing separator.

```suggestion

            string expectedPath = Path.Combine(Directory.GetCurrentDirectory(), directory, name);
            Assert.Equal(expectedPath, args.FullPath);
```

Reply:
fixed